src/System.Data.DataSetExtensions/tests/Mono/EnumerableRowCollectionTest.cs

```diff
@@ -0,0 +1,221 @@
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using Xunit;
+
+namespace MonoTests.System.Data
+{
+    public class EnumerableRowCollectionTest
+    {
+        [Fact]
+        public void QueryWhere()
+        {
+            var ds = new DataSet();
+            ds.ReadXml("testdataset1.xml");
+            var table = ds.Tables[0];
+            /* schema generated as ...
+            var table = ds.Tables.Add ("ScoreList");
+            table.Columns.Add ("ID", typeof (int));
+            table.Columns.Add ("RegisteredDate", typeof (DateTime));
+            table.Columns.Add ("Name", typeof (string));
+            table.Columns.Add ("Score", typeof (int));
+            ds.WriteXml ("Test/System.Data/testdataset1.xsd", XmlWriteMode.WriteSchema);
+            */
+            var q = from line in table.AsEnumerable()
+                    where line.Field<int>("Score") > 80
+                    select line;
+            bool iterated = false;
+            foreach (var line in q)
+            {
+                if (iterated)
+                    Assert.True(false, "should match only one raw");
+                // #1
+                Assert.Equal(100, line["Score"]);
+                iterated = true;
+            }
+        }
+
+        [Fact]
+        public void QueryWhereSelect ()
+        {
+            var ds = new DataSet ();
+            ds.ReadXml ("testdataset1.xml");
+            var table = ds.Tables [0];
+            var q = from line in table.AsEnumerable ()
+            where line.Field<int> ("Score") > 80
+            select new {
+                StudentID = line.Field<int> ("ID"),
+                StudentName = line.Field<string> ("Name"),
+                StudentScore = line.Field<int> ("Score") };
+            bool iterated = false;
+            foreach (var ql in q) {
+            if (iterated)
+                Assert.True(false, "should match only one raw");
+            // #1
+            Assert.Equal(100, ql.StudentScore);
+            iterated = true;
```

Comment:
Nit: the body of the foreach should be indented.  The brace after the foreach should also be on its own line.